src/audio/dai-zephyr.c

```diff
@@ -1602,7 +1602,8 @@ int dai_zephyr_copy(struct dai_data *dd, struct comp_dev *dev, pcm_converter_fun
 	 * if FAST_MODE is enabled, then one period limitation is omitted
 	 */
 	if (!(dd->ipc_config.feature_mask & BIT(IPC4_COPIER_FAST_MODE)))
-		samples = MIN(samples, dd->period_bytes / sampling);
+		samples = MIN(samples,
+			      audio_stream_bytes_to_samples(&buf_c->stream, dd->period_bytes));
```

Comment:
sampling in original code is DMA buffer sample size. Here buf_c is just a temporary variable used to acquire different buffers: DMA buffer or local buffer depending of direction. At this line on code we do not know to what buffer buf_c is pointing. Another thing, is that buf_c was already relesed.